packages/@lightningjs/ui-components/src/mixins/withLongPress/index.js

```diff
@@ -0,0 +1,79 @@
+﻿/**
+ * Copyright 2023 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export default function withLongPress(Base) {
+  return class extends Base {
+    static get name() {
+      return Base.name;
+    }
+    set threshold(value) {
+      this._threshold = value;
+    }
+
+    set targetKey(val) {
+      this._targetKey = val;
+    }
+    set executeOnce(val) {
+      this._executeOnce = val;
+    }
+    _init() {
+      this.pressedTimeStart = null;
+      this.hasExecuted = false;
+      super._init();
+    }
+    /**
+     * this will handle only key down events
+     * it will grab a reference start time stamp and compare it to any subsuquent key down events' timestamp values
+     * when we meet the set threshold time in seconds, we will execute a supplied callback method
+     * */
+    _handleKey(keyEvent) {
+      // capture the key event time stamp thr first time through to use as a reference.
+      if (!this.pressedTimeStart) {
+        this.pressedTimeStart = keyEvent.timeStamp;
+        super._handleKey(keyEvent);
+      }
+      // check latest keyEvent time stamp against the start time stamp and see if the difference
+      // is greater than the threshold
+      if (
+        // eslint-disable-next-line no-constant-condition
+        this.pressedTimeStart &&
+        keyEvent.timeStamp - this.pressedTimeStart > (this._threshold || 2000)
+      ) {
+        // shortcircuit here if we only want to execute once before a key up event
+        if (this._executeOnce && this.hasExecuted) {
+          return;
+        }
+        this.signal('longPressHit', keyEvent.key);
```

Comment:
Would it make sense to change this to a `fireAncestor` call? I can see cases where this would be handled by a grandparent component, but I am also not against keeping this and saying that this is the preferred pattern.

Reply:
Done